WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Container/AddProductWithAIContainerView.swift

```diff
@@ -4,13 +4,17 @@ import SwiftUI
 final class AddProductWithAIContainerHostingController: UIHostingController<AddProductWithAIContainerView> {
     private let viewModel: AddProductWithAIContainerViewModel
     private var addProductFromImageCoordinator: AddProductFromImageCoordinator?
+    private var selectPackageImageCoordinator: SelectPackageImageCoordinator?
 
     init(viewModel: AddProductWithAIContainerViewModel) {
         self.viewModel = viewModel
         super.init(rootView: AddProductWithAIContainerView(viewModel: viewModel))
         rootView.onUsePackagePhoto = { [weak self] productName in
             self?.presentPackageFlow(productName: productName)
         }
+        viewModel.startingInfoViewModel.onPickPackagePhoto = { [weak self] source in
+            await self?.presentImageSelectionFlow(source: source)
+        }
```

Comment:
❓Could you explain why need to use async here? It seems to me that we're mixing the logic of presenting the selection flow with the selected image.

Would it be possible to separate the logic like what we did previously with the old flow? i.e. We keep the presentation of the image picker flow synchronous; and when an image is selected, we trigger a method in `startingInfoViewModel` with the selected image. Would that work?

Reply:
> Could you explain why need to use async here? It seems to me that we're mixing the logic of presenting the selection flow with the selected image.

We are using `async` because we will send the image view into a loading state and wait for the image to be received from this method. Code [here](https://github.com/woocommerce/woocommerce-ios/pull/13193/files#diff-ef2a66020e38bd1b8dd6d9b1980fe144bf3a190f5cf47cf90f38f8314a49463eR56-R68)

> Would it be possible to separate the logic like what we did previously with the old flow? i.e. We keep the presentation of the image picker flow synchronous; and when an image is selected, we trigger a method in startingInfoViewModel with the selected image. Would that work?

I actually followed the pattern from `AddProductFromImageCoordinator` and made this method `async`. We use the same pattern in `BlazeEditAdHostingController` as well. Please correct me if I am misunderstanding things.

Reply:
I'd prefer a more naive implementation for separation of concerns. I'm also worried about the use of `async` because it could potentially come with complications when migrating to Swift 6. Please use your own judgement to decide whether to rewrite this.

Reply:
Thanks for the input, Huong! 

After considering this, I prefer to keep the implementation as is.

My thoughts are,

1. Separation of concerns wise, `SelectPackageImageCoordinator` takes care of selecting and downloading the image. `SelectPackageImageCoordinator` also encapsulates the different paths of camera capture, device media library and WP Media library image selection. After initial selection, all those paths require async operations to prepare the `MediaPickerImage`. i.e. We present the selection flow and wait to get back a `MediaPickerImage`.
1. Using `await` in `ProductCreationAIStartingInfoViewModel` makes it easy to maintain the previous image state and display a loading indicator until the image is ready. 
1. We can alter this if we find any complications while migrating to Swift 6. `SelectPackageImageCoordinator` along with `BlazeEditAdHostingController ` (We will remove `AddProductFromImageCoordinator` while removing package flow code) 

Please let me know if you still think we need to change this. I will add a subtask and work on this at the end of the project. 🙏

Reply:
I'm good with keeping this as-is for now, thanks for the explanation!